src/ai/ai.cpp

```diff
@@ -2735,6 +2800,14 @@ bool will_accept_peace_offer(sys::state& state, dcon::nation_id n, dcon::nation_
 		if(war_duration < 365) {
 			return false;
 		}
+		//The country is ruined and it's been a bit into the war, we'll recover later
+		if(state.world.nation_get_war_exhaustion(n) > 80) {
```

Comment:
these changes will also need to be reflected in the function that calculates the same thing with a bunch of extra parameters (this is used for evaluating a peace offer possibility without actually making it, mostly for the ui)